mitmproxy/io/tnetstring.py

```diff
@@ -226,20 +226,29 @@ def parse(data_type: int, data: bytes) -> TSerializable:
     raise ValueError(f"unknown type tag: {data_type}")
 
 
-def pop(data: bytes) -> tuple[TSerializable, bytes]:
+def split(data: memoryview, sep: bytes) -> tuple[int, memoryview]:
+    i = 0
+    try:
+        while data[i] != ord(sep):
```

Comment:
If I'm not mistaken this will evaluate `ord(sep)` for loop iteration. Do we want to cache it? I'd also be fine with inlining this method below and then hardcoding the separator. :)

Reply:
Personally, I would keep it more modular and therefore keep the split function. You make a good point about "caching" `ord(sep)` !